bigtable/integration_test.go

```diff
@@ -3002,6 +3002,196 @@ func TestIntegration_InstanceUpdate(t *testing.T) {
+		t.Fatalf("Creating backup: %v", err)
+	}
+	wantSourceBackup := srcAdminClient.instancePrefix() + "/clusters/" + srcCluster + "/backups/" + srcBackupName
+
+	destProj1 := testEnv.Config().Project
+	destProj1Inst1 := testEnv.Config().Instance // 1st instance in 1st destination project
+	destProj1Inst1Cl1 := srcCluster             // 1st cluster in 1st instance in 1st destination project
+	destIAdminClient1 := srcIAdminClient
+
+	// Create a 2nd cluster in 1st destination project
+	clusterUID := uid.NewSpace("c-", &uid.Options{Short: true})
+	destProj1Inst1Cl2 := clusterUID.New()
+	defer destIAdminClient1.DeleteCluster(ctx, destProj1Inst1, destProj1Inst1Cl2)
+	err = destIAdminClient1.CreateCluster(ctx, &ClusterConfig{
+		InstanceID:  destProj1Inst1,
+		ClusterID:   destProj1Inst1Cl2,
+		Zone:        instanceToCreateZone2,
+		NumNodes:    1,
+		StorageType: SSD,
+	})
+	if err != nil {
+		t.Fatalf("CreateCluster: %v", err)
+	}
+
+	// Create a 2nd instance in 1st destination project
+	destProj1Inst2, destProj1Inst2Cl1, err := createInstance(ctx, testEnv, destIAdminClient1)
+	if err != nil {
+		t.Fatalf("CreateInstance: %v", err)
+	}
+	defer destIAdminClient1.DeleteInstance(ctx, destProj1Inst2)
+
+	type testcase struct {
+		desc         string
+		destProject  string
+		destInstance string
+		destCluster  string
+	}
+	testcases := []testcase{
+		{
+			desc:         "Copy backup to same project, same instance, same cluster",
+			destProject:  destProj1,
+			destInstance: destProj1Inst1,
+			destCluster:  destProj1Inst1Cl1,
+		},
+		{
+			desc:         "Copy backup to same project, same instance, different cluster",
+			destProject:  destProj1,
+			destInstance: destProj1Inst1,
+			destCluster:  destProj1Inst1Cl2,
+		},
+		{
+			desc:         "Copy backup to same project, different instance",
+			destProject:  destProj1,
+			destInstance: destProj1Inst2,
+			destCluster:  destProj1Inst2Cl1,
+		},
+	}
+
+	if testEnv.Config().Project2 != "" {
+		// Create admin client for 2nd project in test environment
+		destProj2 := testEnv.Config().Project2
+		ctx, options, err := testEnv.AdminClientOptions()
+		if err != nil {
+			t.Fatalf("AdminClientOptions: %v", err)
+		}
+		destIAdminClient2, err := NewInstanceAdminClient(ctx, destProj2, options...)
+		if err != nil {
+			t.Fatalf("NewInstanceAdminClient: %v", err)
+		}
+		defer destIAdminClient2.Close()
+
+		// Create instance in 2nd project
+		destProj2Inst1, destProj2Inst1Cl1, err := createInstance(ctx, testEnv, destIAdminClient2)
+		if err != nil {
+			t.Fatalf("CreateInstance: %v", err)
+		}
+		defer destIAdminClient2.DeleteInstance(ctx, destProj2Inst1)
+		testcases = append(testcases, testcase{
+			desc:         "Copy backup to different project",
+			destProject:  destProj2,
+			destInstance: destProj2Inst1,
+			destCluster:  destProj2Inst1Cl1,
+		})
+	} else {
+		t.Logf("Secondary project not set, skipping copy backup to different project testing")
```

Comment:
not sure about the logging levels here, but a warn level may be nicer if available

Reply:
There is no warning log level available. Added "WARNING" string to log message.